Can't build godot with Visual Studio 2022 due to encoding issue

### Tested versions

Reproducible in 4.3.dev [71699e08c9df78b7203fa4ef9cede28e995d6ace]

### System information

Windows 10 - Visual Studio Community 2022 (64 bits) 17.10.1 - Python 3.11.8 - SCons 4.7.0

### Issue description

When building godot I get this error:
```
1>------ Début de la génération : Projet : godot, Configuration : editor x64 ------
1>Starting SCons
1>**********************************************************************
1>** Visual Studio 2022 Developer Command Prompt v17.10.1
1>** Copyright (c) 2022 Microsoft Corporation
1>**********************************************************************
1>[vcvarsall.bat] Environment initialized for: 'x86_x64'
1>scons: Reading SConscript files ...
1>Auto-detected 4 CPU cores available for build parallelism. Using 4 cores by default. You can override it with the -j argument.
1>Using VCVARS-determined MSVC, arch x86_64
1>Building for platform "windows", architecture "x86_64", target "editor".
1>Checking for C header file mntent.h... (cached) no
1>scons: done reading SConscript files.
1>scons: Building targets ...
[...]
1>progress_finish(["progress_finish"], [])
1>scons: *** [bin\godot.windows.editor.x86_64.exe] UnicodeEncodeError : 'charmap' codec can't encode character '\u251c' in position 5: character maps to <undefined>
1>Traceback (most recent call last):
1>  File "C:\Python311\Lib\site-packages\SCons\Taskmaster\__init__.py", line 246, in execute
1>    self.targets[0].build()
1>  File "C:\Python311\Lib\site-packages\SCons\Node\__init__.py", line 757, in build
1>    self.get_executor()(self, **kw)
1>  File "C:\Python311\Lib\site-packages\SCons\Executor.py", line 386, in __call__
1>    return _do_execute_map[self._do_execute](self, target, kw)
1>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
1>  File "C:\Python311\Lib\site-packages\SCons\Executor.py", line 119, in execute_action_list
1>    status = act(*args, **kw)
1>             ^^^^^^^^^^^^^^^^
1>  File "C:\Python311\Lib\site-packages\SCons\Action.py", line 1349, in __call__
1>    return c.__call__(self, target, source, env, *args, **kw)
1>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
1>  File "C:\Python311\Lib\site-packages\SCons\Action.py", line 1280, in __call__
1>    return act(
1>           ^^^^
1>  File "C:\Python311\Lib\site-packages\SCons\Action.py", line 1521, in __call__
1>    stat = act(target, source, env, exitstatfunc, presub,
1>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
1>  File "C:\Python311\Lib\site-packages\SCons\Action.py", line 724, in __call__
1>    stat = self.execute(target, source, env, executor=executor)
1>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
1>  File "C:\Python311\Lib\site-packages\SCons\Action.py", line 1102, in execute
1>    result = spawn(shell, escape, cmd_line[0], cmd_line, ENV)
1>             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
1>  File "C:\Repos_SSD\_godot_\./platform/windows\detect.py", line 440, in spawn_capture
1>    log.write(line + "\n")
1>  File "C:\Python311\Lib\encodings\cp1252.py", line 19, in encode
1>    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
1>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
1>UnicodeEncodeError: 'charmap' codec can't encode character '\u251c' in position 5: character maps to <undefined>
1>scons: building terminated because of errors.
1>[Time elapsed: 00:02:48.65]
1>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.MakeFile.Targets(45,5): error MSB3073: La commande "echo Starting SCons && cmd /V /C set "plat=x64" ^& (if "x64"=="x64" (set "plat=x86_amd64")) ^& call "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build\vcvars64.bat" !plat! ^& scons --directory="C:\Repos_SSD\_godot_" progress=no platform=windows target=editor arch=x86_64 p=windows debug_symbols=yes" s'est arrêtée avec le code 2.
1>Génération du projet "godot.vcxproj" terminée -- ÉCHEC.
========== Build : 0 réussite(s), 1 échec(s), 0 à jour, 0 ignorée(s) ==========
========== Build s’est terminée à 13:53 et a duré 02:56,579 minutes ==========
```

I can solve the error by replacing "oem" by "utf-8" in [platform/windows/detect.py](https://github.com/godotengine/godot/blob/master/platform/windows/detect.py#L420C54-L420C57), but I am not totally sure that it will work on each platform.
Maybe related to https://github.com/godotengine/godot/pull/89858.

I am not sure it is related to VS 2022, but in case here is my config : [.vsconfig.txt](https://github.com/user-attachments/files/15886422/default.vsconfig.txt)

### Steps to reproduce

- Go to sources of godot
- Call `scons p=windows vsproj=yes debug_symbols=yes` in godot sources root folder
- Open the solution created by previous step with visual studio 2022
- Launch a build
- BUG: When it is finished, one may get an error about encoding


### Minimal reproduction project (MRP)

N/A

--- a/SConstruct
+++ b/SConstruct
@@ -566,7 +566,7 @@ if env["build_profile"] != "":
     import json
 
     try:
-        ft = json.load(open(env["build_profile"]))
+        ft = json.load(open(env["build_profile"], "r", encoding="utf-8"))
         if "disabled_classes" in ft:
             env.disabled_classes = ft["disabled_classes"]
         if "disabled_build_options" in ft:

--- a/editor/template_builders.py
+++ b/editor/template_builders.py
@@ -16,7 +16,7 @@ def parse_template(inherits, source, delimiter):
     meta_prefix = delimiter + " meta-"
     meta = ["name", "description", "version", "space-indent"]
 
-    with open(source) as f:
+    with open(source, "r", encoding="utf-8") as f:
         lines = f.readlines()
         for line in lines:
             if line.startswith(meta_prefix):

--- a/methods.py
+++ b/methods.py
@@ -347,7 +347,7 @@ def detect_modules(search_path, recursive=False):
         # Godot sources when using `custom_modules` build option.
         version_path = os.path.join(path, "version.py")
         if os.path.exists(version_path):
-            with open(version_path) as f:
+            with open(version_path, "r", encoding="utf-8") as f:
                 if 'short_name = "godot"' in f.read():
                     return True
         return False
@@ -989,7 +989,7 @@ def show_progress(env):
             pass
 
     try:
-        with open(node_count_fname) as f:
+        with open(node_count_fname, "r", encoding="utf-8") as f:
             node_count_max = int(f.readline())
     except Exception:
         pass

--- a/platform/macos/SCsub
+++ b/platform/macos/SCsub
@@ -37,7 +37,7 @@ def generate_bundle(target, source, env):
             shutil.copy(target_bin, app_dir + "/Contents/MacOS/Godot")
         version = get_build_version(False)
         short_version = get_build_version(True)
-        with open(Dir("#misc/dist/macos").abspath + "/editor_info_plist.template", "rt") as fin:
+        with open(Dir("#misc/dist/macos").abspath + "/editor_info_plist.template", "rt", encoding="utf-8") as fin:
             with open(app_dir + "/Contents/Info.plist", "wt", encoding="utf-8", newline="\n") as fout:
                 for line in fin:
                     line = line.replace("$version", version)

--- a/platform/windows/detect.py
+++ b/platform/windows/detect.py
@@ -393,7 +393,7 @@ def configure_msvc(env: "SConsEnvironment", vcvars_msvc_config):
 
         # Ensure we have a location to write captured output to, in case of false positives.
         capture_path = methods.base_folder_path + "platform/windows/msvc_capture.log"
-        with open(capture_path, "wt"):
+        with open(capture_path, "wt", encoding="utf-8"):
             pass
 
         old_spawn = env["SPAWN"]
@@ -417,7 +417,7 @@ def configure_msvc(env: "SConsEnvironment", vcvars_msvc_config):
             ret = old_spawn(sh, escape, cmd, args, env)
 
             try:
-                with open(tmp_stdout_name, encoding="oem", errors="replace") as tmp_stdout:
+                with open(tmp_stdout_name, "r", encoding=sys.stdout.encoding, errors="replace") as tmp_stdout:
                     lines = tmp_stdout.read().splitlines()
                 os.remove(tmp_stdout_name)
             except OSError:
@@ -436,7 +436,7 @@ def configure_msvc(env: "SConsEnvironment", vcvars_msvc_config):
                 if not caught and (is_cl and re_cl_capture.match(line)) or (not is_cl and re_link_capture.match(line)):
                     caught = True
                     try:
-                        with open(capture_path, "a") as log:
+                        with open(capture_path, "a", encoding=sys.stdout.encoding) as log:
                             log.write(line + "\n")
                     except OSError:
                         print_warning(f'Failed to log captured line: "{line}".')

